### PR TITLE
Web feeds on global search

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
This is pretty straightforward: it adds the option to render a global search as a web feed, taking inspiration from Reddit. I also added a small edit to the README as installing dependencies from the repository as is will fail if the interpreter is 3.9 or above. I didn't bother to try all interpreter versions down to 3.7, though.

### CODE REVIEW

**Comment by globalistas** [comment]
Can you maybe ensure full diacritics compatibility by using
`a-zA-ZÀ-ž0-9`
instead?

Thanks!

**Comment by why-not-try-calmer** [comment]
> Can you maybe ensure full diacritics compatibility by using `a-zA-ZÀ-ž0-9` instead?
> 
> Thanks!

I do not know what vision the maintainers have about localization, so I don't want to make micro-optimizations without a principled basis. This PR is consistent with the authors' use of regular expressions throughout the code base, and for me that is enough right now.

Of course, one could discuss the use of regexps instead of a more generic approach, and for sure there is something to do about compatibility with diacritics. But I think that's for another time.

**Comment by Polsaker** [comment]
We're planning to replace `A-Za-z` with `\w` in the future, but for now it's OK to use it

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ A phoxy link and discussion aggregator with snek (python3)
 
 ## Setup:
 
-We recommend using a virtualenv or Pyenv
+We recommend using a virtualenv or Pyenv and setting it up to use Python 3.7 for best compatibility (higher versions might fail to build dependencies). See for e.g. [pyenv](https://github.com/pyenv/pyenv) to learn how to install multiple Python versions on your workstation. Refer to your virtual environment manager for documentation on selecting specific Python versions.
```
Comment (Polsaker) [inline README.md:18]: This could be in a separate PR

File: app/views/home.py [hunk 2]
```diff
@@ -212,6 +212,25 @@ def search(page, term):
...
+    posts = misc.getPostList(misc.postListQueryBase(), "new", 1)
+    fg = FeedGenerator()
+    fg.id(request.url)
+    fg.title(f"Search results matching {term}")
+    fg.link(href=request.url_root, rel="alternate")
+    fg.link(href=request.url, rel="self")
+    return Response(
+        misc.populate_feed(fg, posts).atom_str(pretty=True),
+        mimetype="application/atom+xml"
+    )
```
Comment (Polsaker) [inline app/views/home.py:231]: We'll need a config switch to enable/disable this. These searches might be a bit hard on the database on bigger sites (until we implement ElasticSearch or something faster)

**Review by Polsaker — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (why-not-try-calmer) [reply README.md:18]: Will do

File: app/views/home.py (hunk 2 above)
Comment (why-not-try-calmer) [reply app/views/home.py]: Will do

**Comment by Polsaker** [comment]
Looks like the check for `allow_search_feeds` was removed

**Comment by why-not-try-calmer** [comment]
> Looks like the check for `allow_search_feeds` was removed

Thanks, it should be fixed now.
